File: python/24c_151A_SoftDrinking.py
def main():
    n, k, l, c, d, p, nl, np = input().split()
    toasts = (min((int(c) * int(d)), min((int(p) // int(np)), ((int(k) * int(l))// int(nl)))) ) // int(n)
    print(toasts)

if __name__ == "__main__":
    main()

File: cpp/24c_151A_SoftDrinking.cpp
#include<bits/stdc++.h>

int main(){
    int n, k, l, c, d, p, nl, np, toasts;
    std::cin>>n>>k>>l>>c>>d>>p>>nl>>np;
    toasts = (std::min((c * d),std::min((p / np), ((k * l) / nl)))) / n;
    std::cout<<toasts;
    return 0;
}